ktor-client/ktor-client-curl/desktop/src/io/ktor/client/engine/curl/internal/CurlAdapters.kt

```diff
@@ -71,8 +82,10 @@ internal fun HttpRequestData.headersToCurl(): CPointer<curl_slist> {
     var result: CPointer<curl_slist>? = null
 
     mergeHeaders(headers, body) { key, value ->
-        val header = "$key: $value"
-        result = curl_slist_append(result, header)
+        if (isUpgradeRequest() && !DISALLOWED_WEBSOCKET_HEADERS.contains(key)) {
```

Comment:
```suggestion
        if (!isUpgradeRequest() || !DISALLOWED_WEBSOCKET_HEADERS.contains(key)) {
```

Reply:
Otherwise normal requests don't set headers at all

Reply:
Thanks, it was addressed in the recent changes.